src/luascript.cpp

```diff
@@ -4256,6 +4338,33 @@ int LuaScriptInterface::luaSendGuildChannelMessage(lua_State* L)
 	return 1;
 }
 
+
+int LuaScriptInterface::luaLoadDirectory(lua_State* L)
+{
+	const boost::filesystem::path& dir = boost::filesystem::current_path().string() + getString(L, 1);
```

Comment:
Reference to a temporary object (remove the & to fix). Use append() on the path object returned from the current_path() function.
```
const auto dir = boost::filesystem::current_path().append(getString(L, 1));
```